support/ci/shared.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -eou pipefail
+
+#
+# Install core rust toolchain
+# defaults to "stable"
+#
+install_rust_toolchain() {
+  toolchain="${1:-stable}"
+
+  echo "--- :rust: Installing rustup"
+  curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path -y
+  source "$HOME"/.cargo/env
+  echo "--- :rust: Installing rust $toolchain"
+  rustup toolchain install "$toolchain"  
```

Comment:
Similarly:
```bash
  if rustup component list --toolchain "$toolchain" &>/dev/null; then
    echo "--- :rust: rust $toolchain already installed"
  else
    echo "--- :rust: Installing rust $toolchain"
    rustup toolchain install "$toolchain"
  fi
```
This condition is a bit of an odd way to check for a particular toolchain being installed, but I don't know of a better way to leverage `rustup`'s shorthand of `stable` or `nightly`.